src/plugins/vis_type_timeseries/common/last_value_utils.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isArray, last, isEqual } from 'lodash';
+
+export const EMPTY_VALUE = [];
+export const DISPLAY_EMPTY_VALUE = '-';
+
+type ExtractValueFn<T> = (data: T[] | void) => T | null;
+const extractValue: ExtractValueFn<unknown> = (data) => (data && data[1]) ?? null;
+
+type GetLastValueFn<T> = (data: T | T[] | T[][] | void) => T | null;
+export const getLastValue: GetLastValueFn<unknown> = (data) => {
+  if (!isArray(data)) {
+    return data;
+  }
+
+  return extractValue(last(data));
+};
+
+type GetLastValueOrEmptyFn<T, U> = (data: T[][] | void) => T | U;
```

Comment:
usually we are not writing types for each function. It doesn't have a lot of sense because you can do specify all declaration directly for function.
But if for some reason you need a type for this, declare it like this:  `type ExtractValueFn = typeof getLastValueOrEmpty;`

Reply:
I don't really need types for current functions. I'll remove type definitions. I had to change a file extension to ts because of its import to other ts files. Didn't want to leave it in simple js form)